### ISSUE TITLE
Backuppc Buster to Bullseye : error rsync

### ISSUE THREAD

**Issue by FredMa01** [backuppc/backuppc#452]
After migration from buster to bullseye, I have this error :
`Backup failed on 192.168.1.12 (rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0])`

I tried to resolve with this issue #448 but this change nothing.

An other idea ?

Thanks

**Comment by FredMa01** [comment]
It was a problem with SSH port.
I deleted this lines :```

#$Conf{RsyncClientCmd} = '$sshPath -q -x -l root -p 2222 $host $rsyncPath $argList+';
#$Conf{RsyncClientRestoreCmd} = '$sshPath -q -x -l root -p 2222 $host $rsyncPath $argList+';
```
and add in host configuration :
```
$Conf{RsyncSshArgs} = [
  '-e',
  '$sshPath -l root -p2022'
];

```

**Comment by megabert** [comment]
Same Issue here at a new install on Debian Bullseye:

The errors:
```
rsync_bpc: connection unexpectedly closed (0 bytes received so far) [Receiver]
...
rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0]
rsync_bpc exited with fatal status 12 (3072) (rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0])
...
```
The full log of BackupPC_dump -v -f remote-test:
```
Backup type: type = full, needs_full = 1, needs_incr = 1, lastFullTime = 0, opts{f} = 1, opts{i} = , opts{F} = 
cmdSystemOrEval: about to system /bin/ping -c 1 remote-test
cmdSystemOrEval: about to system /bin/ping -c 1 remote-test
CheckHostAlive: ran '/bin/ping -c 1 remote-test'; returning 5.898
XferLOG file /var/lib/backuppc/pc/remote-test/XferLOG.0.z created 2022-01-28 14:31:02 
Backup prep: type = full, case = 6, inPlace = 1, doDuplicate = 0, newBkupNum = 0, newBkupIdx = 0, lastBkupNum = , lastBkupIdx =  (FillCycle = 0, noFillCnt = )
__bpc_progress_state__ backup share "/mnt/webr/a2/87/54416487/htdocs"
Running: /usr/libexec/backuppc-rsync/rsync_bpc --bpc-top-dir /var/lib/backuppc --bpc-host-name remote-test --bpc-share-name /mnt/webr/a2/87/54416487/htdocs --bpc-bkup-num 0 --bpc-bkup-comp 3 --bpc-bkup-prevnum -1 --bpc-bkup-prevcomp -1 --bpc-bkup-inode0 5 --bpc-log-level 9 --bpc-attrib-new -e /usr/bin/ssh\ -l\ root --rsync-path=/usr/bin/rsync --super --recursive --numeric-ids --perms --owner --group -D --times --links --hard-links --delete --delete-excluded --one-file-system --partial --log-format=log:\ %o\ %i\ %B\ %8U,%8G\ %9l\ %f%L --stats --checksum --timeout=72000 remote-test:/mnt/webr/a2/87/54416487/htdocs/ /
full backup started for directory /mnt/webr/a2/87/54416487/htdocs
started full dump, share=/mnt/webr/a2/87/54416487/htdocs
Xfer PIDs are now 40363
xferPids 40363
This is the rsync child about to exec /usr/libexec/backuppc-rsync/rsync_bpc
cmdExecOrEval: about to exec /usr/libexec/backuppc-rsync/rsync_bpc --bpc-top-dir /var/lib/backuppc --bpc-host-name remote-test --bpc-share-name /mnt/webr/a2/87/54416487/htdocs --bpc-bkup-num 0 --bpc-bkup-comp 3 --bpc-bkup-prevnum -1 --bpc-bkup-prevcomp -1 --bpc-bkup-inode0 5 --bpc-log-level 9 --bpc-attrib-new -e /usr/bin/ssh\ -l\ root --rsync-path=/usr/bin/rsync --super --recursive --numeric-ids --perms --owner --group -D --times --links --hard-links --delete --delete-excluded --one-file-system --partial --log-format=log:\ %o\ %i\ %B\ %8U,%8G\ %9l\ %f%L --stats --checksum --timeout=72000 remote-test:/mnt/webr/a2/87/54416487/htdocs/ /
bpc_lib_conf_init: topDir = /var/lib/backuppc, logLevel = 9
bpc_path_create(/var/lib/backuppc/pc/remote-test/0)
bpc_attrib_backwardCompat: WriteOldStyleAttribFile = 0, KeepOldAttribFiles = 0
Connection closed by 81.169.145.126 port 22
rsync_bpc: connection unexpectedly closed (0 bytes received so far) [Receiver]
bpc_sysCall_cleanup: doneInit = 1
RefCnt Deltas for new backup
Uncompressed HT:
Compressed HT:
Done: 0 errors, 0 filesExist, 0 sizeExist, 0 sizeExistComp, 0 filesTotal, 0 sizeTotal, 0 filesNew, 0 sizeNew, 0 sizeNewComp, 5 inode
Parsing done: nFilesTotal = 0
rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0]
rsync_bpc exited with fatal status 12 (3072) (rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0])
Xfer PIDs are now 
xferPids 
Got fatal error during xfer (rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0])
cmdSystemOrEval: about to system /bin/ping -c 1 remote-test
cmdSystemOrEval: about to system /bin/ping -c 1 remote-test
CheckHostAlive: ran '/bin/ping -c 1 remote-test'; returning 5.921
Backup aborted (rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0])
__bpc_progress_state__ fail cleanup
BackupFailCleanup: nFilesTotal = 0, type = full, BackupCase = 6, inPlace = 1, lastBkupNum = 
BackupFailCleanup: inPlace with no new files... no cleanup
__bpc_progress_state__ fsck
Running BackupPC_refCountUpdate -h remote-test -f on remote-test
cmdSystemOrEval: about to system /usr/share/backuppc/bin/BackupPC_refCountUpdate -h remote-test -f
Xfer PIDs are now 40367
xferPids 40367
BackupPC_refCountUpdate: host remote-test #0: gotFsck = 1, gotDelta = , gotPoolCnt = 1, gotNoPoolCntOk = 
BackupPC_refCountUpdate: processing host remote-test #0 (fsck = 1)
__bpc_progress_state__ refCnt #0
__bpc_progress_state__ cntUpdate #0
BackupPC_refCountUpdate: processing host remote-test #0 deltaFile /var/lib/backuppc/pc/remote-test/0/refCnt/poolCntDelta_1_0_0_40367 with 1 entries
__bpc_progress_state__ rename #0
BackupPC_refCountUpdate: computing totals for host remote-test
__bpc_progress_state__ sumUpdate
__bpc_progress_fileCnt__ 0/128
__bpc_progress_fileCnt__ 8/128
__bpc_progress_fileCnt__ 16/128
__bpc_progress_fileCnt__ 24/128
__bpc_progress_fileCnt__ 32/128
__bpc_progress_fileCnt__ 40/128
__bpc_progress_fileCnt__ 48/128
__bpc_progress_fileCnt__ 56/128
__bpc_progress_fileCnt__ 64/128
__bpc_progress_fileCnt__ 72/128
__bpc_progress_fileCnt__ 80/128
__bpc_progress_fileCnt__ 88/128
__bpc_progress_fileCnt__ 96/128
__bpc_progress_fileCnt__ 104/128
__bpc_progress_fileCnt__ 112/128
__bpc_progress_fileCnt__ 120/128
__bpc_progress_state__ rename total
BackupPC_refCountUpdate: host remote-test got 0 errors (took 0 secs)
BackupPC_refCountUpdate total errors: 0
Xfer PIDs are now 
xferPids 
Finished BackupPC_refCountUpdate (running time: 0 sec)
Xfer PIDs are now 
xferPids 
dump failed: rsync error: error in rsync protocol data stream (code 12) at io.c(226) [Receiver=3.1.3.0]
```

The config file:
```
$Conf{BackupPCNightlyPeriod} = 1;
$Conf{XferMethod} = 'rsync';
$Conf{XferLogLevel} = 9;
$Conf{RsyncShareName} = [
  '/mnt/webr/a2/87/54416487/htdocs'
];
```

**Comment by sorin-srbu** [comment]
I ran into this exact problem when trying to backup my debian buster based Proxmox Backup Server (PBS).
Taking Craig Barratt's advice from https://githubmemory.com/repo/backuppc/rsync-bpc/issues/20, I ran the BackupPC_dump from the command line and saw this:

```
...
protected args: --timeout=72000 --numeric-ids . /etc/  (4 args)
bash: line 1: /usr/bin/rsync: No such file or directory
rsync_bpc: connection unexpectedly closed (0 bytes received so far) [Receiver]
bpc_sysCall_cleanup: doneInit = 1 
...
```
Apparently rsync is not included in this particular distro.
Installing it with "apt install rsync" solved the problem for me.
Apart from installing rsync on the client, I did not do anything else.
